Mastodon/Scene/MediaPreview/Video/MediaPreviewVideoViewController.swift

```diff
@@ -90,6 +89,12 @@ extension MediaPreviewVideoViewController {
         }
     }
     
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+
+        playerViewController.didMove(toParent: self)
```

Comment:
The layout changes on the desktop or iPad will raise this method multiple times. Maybe add a precondition check here?

Reply:
Sorry. I try to understand this patch but no answer yet. Is this some UI magics?

@zeitschlag @kimar Please help me and give a review.

Reply:
Yes, you are right. 

From what I noticed during testing: It is indeed some kind of UI magic. The constraints itself are working, however the controls within `AVPlayerViewController` are not correctly set to the safe area insets.
Regarding the lifecycle: it also didn't work in `viewDidAppear`, the earliest trigger I found was `viewDidLayoutSubviews`. As you mentioned, it is run multiple times, but calling `didMove` multiple times wouldn't affect the behavior. It could be guarded with a flag, if you prefer that.
It would also be possible to keep it as it is and delay the `didMove`. Something like:
```
        DispatchQueue.main.asyncAfter(deadline: .now() + 1) { [weak self] in
            self?.playerViewController.didMove(toParent: self)
        }
```
Anyway, please let me know what you prefer and I'll adapt accordingly.

Reply:
I guess the safe area not be applied until the VC attaches to a window. Whatever, it's should an SDK internal issue. If the multiple callings do not break anything. I think the current implementation is great. 

Thank you for your contributions!